runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/Info.java

```diff
@@ -131,11 +78,53 @@ public Info build()
             Info info = new Info(path);
             if (readonly)
             {
-                info.readWorkerCount();
+                try
+                {
+                    byte[] bytes = Files.readAllBytes(info.info);
+                    ByteBuffer byteBuf = ByteBuffer
+                        .wrap(bytes, Long.BYTES, Integer.BYTES)
+                        .order(nativeOrder());
+                    info.workerCount = byteBuf.getInt();
+                }
+                catch (IOException ex)
+                {
+                    System.out.printf("Error: %s is not readable\n", info.info);
+                }
             }
             else
             {
-                info.reset(workerCount);
+                info.workerCount = workerCount;
+                try
+                {
+                    Files.deleteIfExists(info.info);
+                    Files.createDirectories(info.info.getParent());
+                    Files.createFile(info.info);
+
+                    long processId = ProcessHandle.current().pid();
+                    try (SeekableByteChannel channel = Files.newByteChannel(info.info, APPEND))
+                    {
+                        ByteBuffer byteBuf = ByteBuffer
+                                .wrap(new byte[Long.BYTES + Integer.BYTES])
+                                .order(nativeOrder());
+                        byteBuf.putLong(processId);
+                        byteBuf.putInt(workerCount);
+                        byteBuf.flip();
+
+                        while (byteBuf.hasRemaining())
+                        {
+                            channel.write(byteBuf);
+                            Thread.onSpinWait();
+                        }
+                    }
+                    catch (IOException ex)
+                    {
+                        System.out.printf("Error: %s is not writeable\n", info.info);
+                    }
```

Comment:
No need to duplicate this catch block here, the outer one will have the same effect, agree?

Reply:
I removed the inner catch block.